apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/MainActivity.java

```diff
@@ -231,19 +231,23 @@ private void setMaxStorageSize() {
 
             @Override
             public void accept(Boolean succeeded) {
+                final int ALLOWED_SIZE_MULTIPLE = 4096;
                 if (succeeded) {
 
                     /* SQLite always use the next multiple of 4KB as maximum size. */
-                    final int ALLOWED_SIZE_MULTIPLE = 4096;
                     long expectedMultipleMaxSize = (long) Math.ceil((double) maxStorageSize / (double) ALLOWED_SIZE_MULTIPLE) * ALLOWED_SIZE_MULTIPLE;
-
                     Toast.makeText(MainActivity.this, String.format(
                             MainActivity.this.getString(R.string.max_storage_size_change_success),
                             Formatter.formatFileSize(MainActivity.this, expectedMultipleMaxSize)), Toast.LENGTH_SHORT).show();
                     sSharedPreferences.edit().putLong(MAX_STORAGE_SIZE_KEY, expectedMultipleMaxSize).apply();
                 } else {
+
+                    /* SQLite shrinks to fileSize rounded to next page size in that case. */
                     Toast.makeText(MainActivity.this, R.string.max_storage_size_change_failed, Toast.LENGTH_SHORT).show();
-                    sSharedPreferences.edit().putLong(MAX_STORAGE_SIZE_KEY, 0).apply();
+                    String DATABASE_NAME = "com.microsoft.appcenter.persistence";
+                    long fileSize = getDatabasePath(DATABASE_NAME).length();
+                    long maxSize = (long) Math.ceil((double) fileSize / (double) ALLOWED_SIZE_MULTIPLE) * ALLOWED_SIZE_MULTIPLE;
```

Comment:
This is redundant since you are reading an actual size. It will always be incremental of 4K.

Reply:
Sure, will simplify code.